packages/components/src/focal-point-picker/test/index.js

```diff
@@ -62,6 +62,34 @@ describe( 'FocalPointPicker', () => {
 		} );
 	} );
 
+	describe( 'resolvePoint handling', () => {
+		it( 'should allow value altering', async () => {
+			const spyChange = jest.fn();
+			const spy = jest.fn();
+			const { getByRole } = render(
+				<Picker
+					value={ { x: 0.25, y: 0.25 } }
+					onChange={ spyChange }
+					resolvePoint={ () => {
+						spy();
+						return { x: 0.91, y: 0.42 };
+					} }
```

Comment:
Q: when checking this test on my local machine, I tried to make the test fail by modifying the `resolvePoint` function to

```jsx
resolvePoint={ ( value ) => {
  spy();
  return value;
} }
```

The test fails (as expected), but the values received by the `onChange` spy are not `x: 0.25, y: 0.25`, but instead are `x: 0, y: 0`. I wasn't expecting this behaviour, do you know why would this happen?

Reply:
My guess is that the main reason is that the picker doesn't have width and height props. Also, fireEvent is called without any other parameters but am not sure if the case would be the same if the picker had set dimensions.

Reply:
Ok, so I haven't been able to use `mouseMove` to create a proper emulation of a user interaction. However, I managed that by using `fireEvent.keyDown( dragArea, { charCode: 0, keyCode: 38 } );` which triggers onChange with an "expected" value of `x: 0.25, y: 0.24` making my test fail. That was because onChange was called with the same value `updateValue` would received. 

I added a second parameter to the `updateValue` method which is a callback that is passed to the setState call. Moving the `onChange` call inside that callback fixed the issue and made the test pass. Everything looks fine now.

Reply:
I'm not sure I understand what you're asking for. Doesn't the initial test cover this behaviour? Do you want me to remove the onChange part from that test and then make another one in the `controllability` section that only tests the onChange part?

Reply:
I'm proposing to add another unit test aimed at testing the behaviour that I highlighted earlier and you fixed in your latest commits by using the `setState` callback to call `onChange`.

Earlier, when removing `resolvePoint` from this test, the `onChange` spy was receiving `x: 0, y: 0` instead of the expected value. Thanks to your fix, the `onChange` would now receive the correct values. My proposal is to add an additional unit test to specifically test that (note that, in the new test, the `resolvePoint` prop is not specified)

Hope I managed to explain myself better :)

Reply:
I'll add the unit test but I want to make sure we're on the same page.

By using `mouseDown` and `mouseMove` we still receive { x: 0.00, y: 0.00 } instead of the values with which the picker is initialised.

The bug that was fixed by moving the onChange inside the setState callback was related to the `resolvePoint` prop.
```js
this.updateValue( next ); // next was being passed to resolvePoint here
this.props.onChange( next ); // but not here
```

This unit test becomes a test to future proof the keyboard interaction (rather for the UP key press). Am I right?

Reply:
> By using mouseDown and mouseMove we still receive { x: 0.00, y: 0.00 } instead of the values with which the picker is initialised.

I see, I missed that detail. Do you think it's worth investigating this in a follow-up PR?